lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -351,8 +390,32 @@ static bool validateCodingKeysEnum(const DerivedConformance &derived,
     }
   }
 
-  if (!varDeclsAreValid)
+  if (!varDeclsAreValid) {
+    for (auto&& elt : extraKeys) {
+      Optional<Identifier> fixit = None;
+      unsigned bestScore = 0;
+      for (auto&& varDecl : varDecls) {
+        auto score = scoreParamAndArgNameTypo(elt->getBaseIdentifier().str(), varDecl.first.str(), bestScore);
+        if (score.hasValue() && (!fixit.hasValue() || score.getValue() < bestScore)) {
+          fixit = varDecl.first;
+          bestScore = score.getValue();
+        }
+      }
+      if (fixit.hasValue()) {
+        auto suggestion = fixit.getValue();
+        delayedNotes.push_back([=] {
+          elt->diagnose(diag::codable_extraneous_codingkey_case_here,
```

Comment:
Let's issue a custom diagnostic here. Perhaps

```
CodingKey case %0 nearly matches stored property %1; did you mean to use %1 instead?
```

Along with `descriptive_generic_type_declared_here` on the stored property